src/ngx_http_lua_pipe.c

```diff
@@ -0,0 +1,2486 @@
+                              "lua pipe child " ngx_close_socket_n
+                              " %V failed", &ls[i].addr_text);
+            }
+        }
+
+        /* close and dup pipefd */
+        if (close(in[1]) == -1) {
+            ngx_log_error(NGX_LOG_EMERG, ngx_cycle->log, ngx_errno,
+                          "lua pipe child failed to close the in[1] "
+                          "pipe fd");
+        }
+
+        if (close(out[0]) == -1) {
+            ngx_log_error(NGX_LOG_EMERG, ngx_cycle->log, ngx_errno,
+                          "lua pipe child failed to close the out[0] "
+                          "pipe fd");
+        }
+
+        errlog_fd = ngx_cycle->log->file->fd;
+        if (errlog_fd == STDERR_FILENO) {
+            temp_errlog_fd = dup(errlog_fd);
+
+            if (temp_errlog_fd == -1) {
+                ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                              "lua pipe child dup errlog fd failed");
+                exit(EXIT_FAILURE);
+            }
+
+            if (ngx_cloexec(temp_errlog_fd) == -1) {
+                ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                              "lua pipe child new errlog fd " ngx_cloexec_n
+                              " failed");
+            }
+
+            ngx_log_debug2(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
+                           "lua pipe child dup old errlog fd %d to new fd %d",
+                           ngx_cycle->log->file->fd, temp_errlog_fd);
+
+            ngx_cycle->log->file->fd = temp_errlog_fd;
+        }
+
+        if (dup2(in[0], STDIN_FILENO) == -1) {
+            ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                          "lua pipe child dup2 stdin failed");
+            exit(EXIT_FAILURE);
+        }
+
+        if (dup2(out[1], STDOUT_FILENO) == -1) {
+            ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                          "lua pipe child dup2 stdout failed");
+            exit(EXIT_FAILURE);
+        }
+
+        if (merge_stderr) {
+            if (dup2(STDOUT_FILENO, STDERR_FILENO) == -1) {
+                ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                              "lua pipe child dup2 stderr failed");
+                exit(EXIT_FAILURE);
+            }
+
+        } else {
+            if (close(err[0]) == -1) {
+                ngx_log_error(NGX_LOG_EMERG, ngx_cycle->log, ngx_errno,
+                              "lua pipe child failed to close the err[0] "
+                              "pipe fd");
+            }
+
+            if (dup2(err[1], STDERR_FILENO) == -1) {
+                ngx_log_error(NGX_LOG_ERR, ngx_cycle->log, ngx_errno,
+                              "lua pipe child dup2 stderr failed");
+                exit(EXIT_FAILURE);
+            }
+        }
+
+        ngx_log_debug2(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
+                       "lua pipe spawn process:%s pid:%P", file, getpid());
```

Comment:
One of the two "spawn" debug log should be removed. They currently produce a duplicated output:
```
2019/01/14 11:44:03 [debug] 9552#0: lua pipe spawn process:0000000041529F38 pid:9553 merge_stderr:0 buffer_size:4096
2019/01/14 11:44:03 [debug] 9552#0: *1 lua pipe read process:0000000041529F38 pid:9553
2019/01/14 11:44:03 [debug] 9552#0: lua pipe init pipe ctx:000000000172CFC0 fd:*12
2019/01/14 11:44:03 [debug] 9552#0: lua pipe spawn process:openssl pid:9553
```

imho, we should keep the one logged from the worker process. Not only is it more detailed, but it is also safer not to log the `file` being executed.